all-of-us/workbench
# [RW-9302][risk=no]   [API][Cromwell] Install and config Cromshell in AoU Jupyter image

The purpose of this PR is to ensure when jupyter starts : cromshell is installed and is pointing to the server created by Leonardo team for the specific user and workspace.

Workbench provides appName to leonardo team when creating an app, hence we save the name in DB to ensure jupyter and cromwell app are in sync, so that irrespective of which app or runtime runs first, the user can run workflow using the correct cromwell server information.

The one edge case where this solution will not work is : Jupyter is started and user deletes and creates a new cromwell server. In this case jupyter will still have old information. 
Workaround: user will have to refresh jupyter runtime.

How to test:
1) Deploy your own server using the following command: 
./project.rb deploy-api --no-promote --version "${USER}" --project all-of-us-workbench-test

2) Change the following entry in config_local : "server" -> "apiBaseUrl" to point the server created in step 1 (eg     "apiBaseUrl": "https:\/\/nsaxena-dot-api-dot-all-of-us-workbench-test.appspot.com")

3) Restart notebook server 


Check: main/webapp/static/README.md



Description:



---
**PR checklist**

- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert
